packages/core/src/browser/keybinding.ts

```diff
@@ -556,7 +556,9 @@ export class KeybindingRegistry {
                         const commandHandler = this.commandRegistry.getActiveHandler(command.id);
 
                         if (commandHandler) {
-                            commandHandler.execute();
+                            const commandArgs = binding.args || [];
+                            const commandArgsArr = commandArgs instanceof Array ? commandArgs : [commandArgs];
```

Comment:
1. Such args should be applied to retrieve active, visible and toggled handler in `CommandRegsitry` as well as in all places where a command is retrieved via them and executed.
2. Are you sure that it is aligned with how VS Code handles args? For some reasons they assign `any` as a type to `args`. It looks like they don't do any conversions but simply pass it to `execute` method. The best approach to verify will be to create a custom command via VS Code extension, pass to it different args and print them. Then this extension can be shared with other committers for double testing.

Reply:
Agree with point 1 as I didn't notice that I'll update it to do that.

In regards to point 2 this was to handle the way task 'args' work in VS Code although it says args its actually just the label of the task
```
{
    "key": "ctrl+h",
    "command": "workbench.action.tasks.runTask",
    "args": [
      "echo"
    ]
}
```
If you were to try to do this in VSCode, it just asks you what task you want to run as its expecting args to just be a string not an array.

 This is just me converting the string into an array. Although it is probably better suited lower down when parsing then here.

Reply:
We should align with point 2 for tasks as well, so it seems that we should drop conversions, type of `args` should be `any` and we just pass it to `execute` as it is.

> it just asks you what task you want to run as its expecting args to just be a string not an array. (If you pick the task you want to run these arguments aren't actually passed down to the task)

our `workbench.action.tasks.runTask` should do the same

Reply:
Cool agreed, so current functionality is that if it can't find the task for any reason (including not including 'args' it being an array or just the value not being found) it will show the quick menu to pick a task.

Otherwise the task is executed as normal. I've also changed getting the command handler to pass in the args and now there is no type conversion when passed to the command handler its just passed in. - Let me know if any other changes are needed.